code/modules/power/supermatter/supermatter.dm

```diff
@@ -221,7 +221,7 @@
 		return
 	var/mob/living/carbon/human/H = user
 	var/immune = istype(H.glasses, /obj/item/clothing/glasses/meson)
-	if(!immune && (get_dist(user, src) < HALLUCINATION_RANGE(power)))
+	if(!immune && !HAS_TRAIT(H, TRAIT_MESON_VISION) && (get_dist(user, src) < HALLUCINATION_RANGE(power)))
```

Comment:
Don't suppose you could change meson goggles to just add the `TRAIT_MESON_VISION` trait on equip? That way you can get rid of the `istype(x.glasses, /obj/item/clothing/glasses/meson)` checks in this file.

Reply:
thats gonna be out of scope, it requires a change to how glasses work